### PR TITLE
allow to run multiple plugins

### PR DESCRIPTION
**Pull request by ksimon1**
This PR allows to labeller to run multiple plugins

### CODE REVIEW

File: cmd/cpu-node-labeller/cpu-node-labeller.go [hunk 1]
```diff
@@ -78,13 +67,42 @@ func main() {
...
+}
+
+func runFile(fileDir, fileName string, features map[string]string) error {
+	path := filepath.Join(fileDir, fileName)
+	filestat, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if filestat.Mode().IsRegular() {
```
Comment (fromanirh) [inline cmd/cpu-node-labeller/cpu-node-labeller.go]: maybe check that IsRegular() && is executable?

**Comment by fromanirh** [comment]
Do we need to generalize the label removal?

**Comment by ksimon1** [comment]
Good catch, i will update it.

File: pkg/node/node.go [hunk 2]
```diff
@@ -51,10 +51,10 @@ func AddNodeLabels(node *v1.Node, labels map[string]string) {
 	}
 }
 
-// RemoveCPUModelNodeLabels removes labels from node with prefix: feature.node.kubernetes.io/cpu-model-*
+// RemoveCPUModelNodeLabels removes labels from node with prefix: feature.node.kubernetes.io/*
 func RemoveCPUModelNodeLabels(node *v1.Node) {
 	for label := range node.Labels {
-		if strings.Contains(label, labelNamespace+"cpu-model-") {
+		if strings.Contains(label, labelNamespace) {
```
Comment (fromanirh) [inline pkg/node/node.go]: this assumes our cpu-node-labeller owns the labels of the node (e.g. noone else can add feature.node.kubernetes.io/ labels). Is that right?

File: cmd/cpu-node-labeller/cpu-node-labeller.go [hunk 3]
```diff
@@ -78,13 +67,42 @@ func main() {
...
+		// Run hook
+		err = cmd.Run()
+
+		lines := bytes.Split(stdout.Bytes(), []byte("\n"))
+		for _, feature := range lines {
+			if len(feature) == 0 {
+				continue
+			}
+
+			features[string(feature)] = "true"
```
Comment (fromanirh) [inline cmd/cpu-node-labeller/cpu-node-labeller.go]: OK for the time being. In the future we may want to allow negative labels.

Comment (ksimon1) [reply cmd/cpu-node-labeller/cpu-node-labeller.go]: In the future, we will be using openshift NFD :)

File: pkg/node/node.go (hunk 2 above)
Comment (ksimon1) [reply pkg/node/node.go]: NFD is using same syntax. So if someone using nfd and this labeller, then labeller deletes all labels.

Comment (MarSik) [reply pkg/node/node.go]: This is not a good idea. We want to be able to use our labeller together with NFD. The idea is to move to NFD once it is clear how our plugins should be integrated, but we should not stomp on NFD's feet.

Can we fix this somehow? NFD uses an annotation where it stores all the labels it set.

File: cmd/cpu-node-labeller/cpu-node-labeller.go (hunk 3 above)
Comment (MarSik) [reply cmd/cpu-node-labeller/cpu-node-labeller.go]: Negative labels could be represented by not adding the label at all :)

File: pkg/node/node.go (hunk 2 above)
Comment (ksimon1) [reply pkg/node/node.go]: yes, i was thinking about doing it like NFD.

Comment (ksimon1) [reply pkg/node/node.go]: should be fixed now

Comment (fromanirh) [reply pkg/node/node.go]: Let's try to add unit/functional tests if feasible.

**Pull request merged**